File: src/engine/sprite/simple-sprite.lua
---@class SimpleSprite:Sprite
---@field width number
---@field height number
---@field image love.Image
SimpleSprite = setmetatable({}, { __index = Sprite })
SimpleSprite.__index = SimpleSprite

---@param image love.Image
function SimpleSprite:new(image)
    local obj = {
        width = image:getWidth(),
        height = image:getHeight(),
        image = image
    }
    setmetatable(obj, SimpleSprite)
    return obj
end

---@param x number
---@param y number
function SimpleSprite:draw(x, y)
    love.graphics.draw(self.image, x, y)
end

return SimpleSprite

File: src/engine/sprite/simple-sprite.ts
import { Image } from "love.graphics";
import { Vector2 } from "../geometry/vector2";
import { Sprite } from "./sprite";

export class SimpleSprite implements Sprite {
  constructor(private image: Image) {
    this.image = image;
  }
  getWidth(): number {
    return this.image.getWidth();
  }
  getHeight(): number {
    return this.image.getHeight();
  }

  draw(pos: Vector2) {
    love.graphics.draw(this.image, pos.x, pos.y);
  }

  update(): void {
    // do nothing
  }
}
